### ISSUE TITLE
Using UUIDs instead of ObjectIDs

### ISSUE THREAD

**Issue by chrisie** [LearnBoost/mongoose#2738]
I am trying to replace the ObjectIDs used for the _id attributes of my documents withUUIDs. I am generating my UUIDs with node-uuid (uuid.v4()) and in order to save space I am also converting them to binary before passing them to the DB (with uuid.parse()). From what I've read I understand that the correct way to store these UUIDs in MongoDB would be to store them as a Binary type with the UUID subtype. I have therefore written the following:

```
var parentSchema = mongoose.Schema({
  _id: {type: Buffer, required: true, default: function () {
    return mongoose.Types.Buffer(uuid.parse(uuid.v4())).toObject(0x04);
  }},
children: [childSchema]
});
```

This works fine, however when I attempt to set the _id of the childSchema the same way I get the following error: 

```
this.__parentArray._markModified();
                       ^
TypeError: undefined is not a function
    at EmbeddedDocument.markModified (/my-project/node_modules/mongoose-promised/node_modules/mongoose/lib/types/embedded.js:62:24)
...
```

Is this not the way I am supposed to be storing UUIDs or am I missing something else regarding how mongoose handles _ids?

**Comment by vkarpov15** [comment]
Hmm not sure, I'll investigate this. Can you let me know what version of mongoose you're using?

**Comment by chrisie** [comment]
I'm using the latest available version of mongoose (3.8.24).
